### DISCUSSION TITLE
Modify stack_output value in stack config

### DISCUSSION THREAD

**Discussion by cornerman** [cloudreach/sceptre#752] [Q&A]
Is it possible to modify the output value of stack within a stack config?

My use-case is that I want to use the `Arn` of an S3-Bucket, which comes out of another stack, but want to append `/*` to the arn to specify a path within the bucket. I want to pass this value into another stack as a parameter. I am doing something like this in my stack.yaml:

```yaml
parameters:
  BucketArn:  !stack_output group/s3-bucket.yaml::BucketArn # I would like to append /* here
```

I have tried ` !stack_output group/s3-bucket.yaml::BucketArn{{ "/*" }}` and `(!stack_output group/s3-bucket.yaml::BucketArn)/*`, but that does not work.

Is that possible?

relevant: https://github.com/cloudreach/sceptre/issues/735

**Comment by ngfgrant** [comment]
Probably need to use a CusomResolver for this unless there are any jinja tricks @1oglop1 ?

**Comment by 1oglop1** [comment]
No, you are right. This is the reason we developed SmartJoin resolver.
It would be nice if we could actually find some time and publish package with our hooks and resolvers.

I'm thinking of creating a new repo/s under Sceptre org. 
My idea is to put all resolvers and hooks we have into 1 package and publish it to pypi.

Then maintain it as a whole set instead micro packages for each resolver which has under 100 lines of code.

The repo could also serve as a best practice example instead of current brief example in docs which is a bit misleading for the newcomers to sceptre and python.

Or implement support for custom jinja filters to give advanced users more flexibility.

**Comment by ngfgrant** [comment]
@1oglop1 Yeh that is definitely one of the benefits that the new org will bring us. At the moment there are some empty hooks/resolver template repos which I still need to push some local code to. Cloning them will give you everything you need just to write you resolver/hook code and push the build and deploy will be handled for you. 

If you are able to even push what you have to the new org then I can take some time to put the remaining boiler plate in place?

**Comment by cornerman** [comment]
Nice, that sounds awesome. Then I am really looking forward to the `SmartJoin` Resolver :)

**Comment by 1oglop1** [comment]
@ngfgrant I'd rather provide `cookiecutter` version of hooks and resolver 
https://github.com/audreyr/cookiecutter
instead of letting it fork/clone.

**Comment by ngfgrant** [comment]
I think GitHub template repository feature is probably enough for what we need (https://help.github.com/en/articles/creating-a-template-repository). I mean we could easily integrate cloning the template repo in the `sceptre new ...` command, that would be a pretty straightforward helper. 

`sceptre new hook HOOK_NAME`
    -> git clones the hook template repo into directory called `hook_name`
    -> search and replace in correct places with `hook_name`
    -> print "Your new hook project is available at `./hook_name`"

**Comment by jghaines** [comment]
For the original case, can you not also create a separate Output value and use `!Sub` to append to it?

**Comment by cornerman** [comment]
> For the original case, can you not also create a separate Output value and use !Sub to append to it?

@jghaines Thank you for the idea. That would work, but it does not feel right! The template having the outputs should not have to know about all possible usage-scenarios. In this case, it just creates an s3-bucket; and should not need to prepare for all possible paths I want to append to the bucket-name. I think, the transformation/substitution should be done on the usage-side.

**Comment by haseebr** [comment]
I've been working on a resolver that is able to pipe outputs of one resolver into the next one, similar to how it's done in shell scripting.
It would look something like this

```
parameters:
    DBPassword: !pipe
        - !aws secretsmanager::get_secret_value::'SecretId':'super-secret-value'::SecretString
        - !cmd echo '(( result ))' | jq -r ". | .password"
```

Just got to find time to clean it up and publish it.

**Comment by 1oglop1** [comment]
Interesting idea.
All these workarounds are really smelly compared to native functionality of `jinja filters`
